src/main/java/lotto/model/Cashier.java

```diff
@@ -0,0 +1,30 @@
+package lotto.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Cashier {
+
+    private static final int LOTTO_PURCHASE_AMOUNT_UNIT = 1_000;
+
+    public List<Lotto> payForLotto(int amount) {
+        int count = calculateLottoCount(amount);
+        return issueLottoByCount(count);
+    }
+
+    private List<Lotto> issueLottoByCount(int count) {
+        List<Lotto> lottos = new ArrayList<>();
+        for (int i = 0; i < count; i++) {
+            lottos.add(LottoMachine.issue());
+        }
+        return List.copyOf(lottos);
+    }
+
+    private int calculateLottoCount(int amount) {
+        if (amount % LOTTO_PURCHASE_AMOUNT_UNIT != 0) {
+            String message = String.format("로또는 %,d원 단위로 구매할 수 있습니다.", LOTTO_PURCHASE_AMOUNT_UNIT);
+            throw new IllegalArgumentException(message);
```

Comment:
변수 생성은 최소화 하는게 좋다고 생각해요!
나중에 GC의 대상이 됩니다 😃 

꼭 필요한 경우가 아니면 변수를 할당하지 않는게 좋다고 생각해요!

```suggestion
            throw new IllegalArgumentException(String.format("로또는 %,d원 단위로 구매할 수 있습니다.", LOTTO_PURCHASE_AMOUNT_UNIT));
```

Reply:
메세지의 formatting 때문에 코드 한 줄이 너무 길어져 가독성에 영향을 줄 것이라고 생각했어요.
그런데 GC 대상이 될 것이란 건 생각지 못했네요 😲
중요한 역할이거나 필요한 상황이 아니라면 변수 생성은 줄이는 게 좋겠군요!
다음 미션 때 고려해서 진행해보겠습니다 !! :)